modin/engines/ray/utils.py

```diff
@@ -135,9 +137,24 @@ def initialize_ray(
             object_store_memory = Memory.get()
             # In case anything failed above, we can still improve the memory for Modin.
             if object_store_memory is None:
-                # Round down to the nearest Gigabyte.
-                system_memory = ray._private.utils.get_system_memory()
-                object_store_memory = int(0.6 * system_memory // 10 ** 9 * 10 ** 9)
+                virtual_memory = psutil.virtual_memory().total
+                if sys.platform.startswith("linux"):
+                    shm_fd = os.open("/dev/shm", os.O_RDONLY)
+                    try:
+                        shm_stats = os.fstatvfs(shm_fd)
+                        system_memory = shm_stats.f_bsize * shm_stats.f_bavail
+                        if system_memory / (virtual_memory / 2) < 0.99:
```

Comment:
```suggestion
                        if system_memory < (virtual_memory / 2):
```
I think this way it's easier to understand :)

Reply:
Should we add this hard condition on size of `system_memory` (/dev/shm size)? Often, different small-size files from apps are remained in /dev/shm. If we add your condition we must free /dev/shm before each Modin's using. What do you think?

Reply:
I don't think it would be correct to free /dev/shm. That's quite destructive to the user's environment.

Reply:
I suggest leaving it as it is. In this case we don't add hard condition on half of virtual memory and this gives a small space to user's files in /dev/shm.

Reply:
you can keep `0.99` multiplier, it's just comparing `A` to `B` is easier to read than `A / B` to `1` IMHO.